src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -650,33 +749,65 @@ void StorageObjectStorage::Configuration::check(ContextPtr) const
     FormatFactory::instance().checkFormatName(format);
 }
 
-bool StorageObjectStorage::Configuration::withPartitionWildcard() const
+StorageObjectStorage::Configuration::Path StorageObjectStorage::Configuration::getReadingPath() const
+{
+    auto raw_path = getRawPath();
+
+    if (!partition_strategy)
+    {
+        return raw_path;
+    }
+
+    return Path {partition_strategy->getReadingPath(raw_path.path)};
```

Comment:
Why not initialize reading path once - when raw path is set? to avoid recalculating reading path each time we need it. Same about writing path

Reply:
Done